packages/amplify-graphql-relational-transformer/src/belongs-to/belongs-to-directive-transformer-factory.ts

```diff
@@ -21,18 +21,22 @@ export const getBelongsToDirectiveTransformer = (
     case 'POSTGRES':
       return belongsToDirectivePostgresTransformer;
     case 'DYNAMODB':
-      // If `references` are passed to the directive, we'll use the references relational
+    // If references are passed to the directive, we'll use the references relational
       // modeling approach.
-      if (config.references?.length >= 1) {
-        // Passing both `references` and `fields` is not supported.
-        if (config.fields?.length > 0) {
+      if (config.references) {
+        // Passing both references and fields is not supported.
+        if (config.fields) {
```

Comment:
nit: would avoid this nesting especially for a known failure condition to improve readability.

Reply:
This seems like a schema validation rather than something we will run on a `getTransformer` call. Any thoughts here?

Reply:
The suggestion is to do validation (`if (config.references && config.fields) { // error }`) in the `validate` method, but still branch here to determine if we should use `references` or `fields` based behavior?

That sounds good. I'll include that in a subsequent PR that handles this open item being tracked in the [feature branch PR](https://github.com/aws-amplify/amplify-category-api/pull/2370)
>  Move all necessary validation to validate methods in relational directive transformers.

Reply:
Yes, that would be nice.